src/main/java/com/sourcery/gymapp/backend/sharedLinks/model/BaseEntity.java

```diff
@@ -0,0 +1,44 @@
+package com.sourcery.gymapp.backend.sharedLinks.model;
+
+
+import jakarta.persistence.MappedSuperclass;
+import jakarta.persistence.EntityListeners;
+import jakarta.persistence.Id;
+import jakarta.persistence.Column;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.data.annotation.CreatedBy;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedBy;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+@Getter
+@Setter
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener.class)
+public class BaseEntity {
+
+    @Id
+    @Column(updatable = false, nullable = false)
```

Comment:
My bad if I commented the wrong line, but I meant to say that the @GeneratedValue(strategy = GenerationType.UUID) should be here, and @Column not